gui/src/daemon/client/jsonrpc.rs

```diff
@@ -97,12 +88,14 @@ impl JsonRPCClient {
 
         debug!("Sending to lianad: {:#?}", request);
 
-        stream.write_all(&[serde_json::to_string(&request).unwrap().as_bytes(), b"\n"].concat())?;
+        stream
+            .write_all(&[serde_json::to_string(&request).unwrap().as_bytes(), b"\n"].concat())
+            .await?;
+
+        let mut buffer = Vec::<u8>::new();
+        stream.read_to_end(&mut buffer).await?;
```

Comment:
When testing with an external daemon, I found it was hanging on this step (calling `getinfo`) while loading the GUI.

I did a quick test adding `stream.shutdown().await?;` just before reading and that seemed to fix it, but not sure if that's the correct way.